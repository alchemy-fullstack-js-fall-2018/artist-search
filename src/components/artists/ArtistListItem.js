import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import PropTypes from 'prop-types';

const ArtistListItem = ({ result }) => {
  const { name, type, description, id } = result;

  return (
    <li>
      <p>{name} ({type}) - {description}</p>
      <Link to={ROUTES.ARTIST.linkTo(id)}>Get Details</Link>
    </li>
  );
};

ArtistListItem.propTypes = {
  result: PropTypes.object.isRequired
};

export default ArtistListItem;
