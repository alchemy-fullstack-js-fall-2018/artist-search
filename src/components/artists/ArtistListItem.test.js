import React from 'react';
import { shallow } from 'enzyme';
import ArtistListItem from './ArtistListItem';

describe ('tests for ArtistListItem component', () => {

  it('renders an ArtistListItem component', () => {
    const result = {
      name: 'Joni Mitchell',
      type: 'Person',
      description: 'American folk singer',
      id: '12345'

    };
    const wrapper = shallow(<ArtistListItem result={result}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
