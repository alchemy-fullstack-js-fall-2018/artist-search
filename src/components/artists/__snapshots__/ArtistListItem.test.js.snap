// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for ArtistListItem component renders an ArtistListItem component 1`] = `
<li>
  <p>
    Joni Mitchell
     (
    Person
    ) - 
    American folk singer
  </p>
  <Link
    replace={false}
    to="/artists/12345"
  >
    Get Details
  </Link>
</li>
`;
